src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+enum Shapes {
+  circle = 'circle',
+  triangle = 'triangle',
+  rectangle = 'rectangle',
+}
+
+enum Colors {
+  red = 'red',
+  blue = 'blue',
+  green = 'green',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shapes.triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign them to `this` as you have done on line 20.